packages/solid/src/use-dotlottie.tsx

```diff
@@ -0,0 +1,306 @@
+  };
+
+  onCleanup(() => {
+    dotLottie()?.destroy();
+    setDotLottie(null);
+    resizeObserver()?.disconnect();
+    intersectionObserver()?.disconnect();
+    if (canvasRef) {
+      canvasRef.removeEventListener('mouseenter', hoverHandler);
+      canvasRef.removeEventListener('mouseleave', hoverHandler);
+    }
+  });
+
+  // Reactivities
+  // speed
+  createEffect(() => {
+    const dotLottieInstance = dotLottie();
+
+    if (!dotLottieInstance) return;
+
+    if (typeof config.speed === 'number' && config.speed !== dotLottieInstance.speed && dotLottieInstance.isLoaded) {
+      dotLottieInstance.setSpeed(config.speed);
+    }
+  });
+  // mode
+  createEffect(() => {
+    const dotLottieInstance = dotLottie();
+
+    if (!dotLottieInstance) return;
+
+    if (typeof config.mode === 'string' && config.mode !== dotLottieInstance.mode && dotLottieInstance.isLoaded) {
+      dotLottieInstance.setMode(config.mode);
+    }
+  });
+  // loop
+  createEffect(() => {
+    const dotLottieInstance = dotLottie();
+
+    if (!dotLottieInstance) return;
+
+    if (typeof config.loop === 'boolean' && config.loop !== dotLottieInstance.loop && dotLottieInstance.isLoaded) {
+      dotLottieInstance.setLoop(config.loop);
+    }
+  });
+  // useFrameInterpolation
+  createEffect(() => {
+    const dotLottieInstance = dotLottie();
+
+    if (!dotLottieInstance) return;
+
+    if (
+      typeof config.useFrameInterpolation === 'boolean' &&
+      config.useFrameInterpolation !== dotLottieInstance.useFrameInterpolation &&
+      dotLottieInstance.isLoaded
+    ) {
+      dotLottieInstance.setUseFrameInterpolation(config.useFrameInterpolation);
+    }
+  });
+  // segment
+  createEffect(() => {
+    const dotLottieInstance = dotLottie();
+
+    if (!dotLottieInstance) return;
+
+    if (typeof config.segment === 'object' && Array.isArray(config.segment) && dotLottieInstance.isLoaded) {
+      const startFrame = config.segment[0];
+      const endFrame = config.segment[1];
+
+      dotLottieInstance.setSegment(startFrame, endFrame);
+    }
+  });
+  // background color
+  createEffect(() => {
+    const dotLottieInstance = dotLottie();
+
+    if (!dotLottieInstance) return;
+
+    if (
+      typeof config.backgroundColor === 'string' &&
+      config.backgroundColor !== dotLottieInstance.backgroundColor &&
+      dotLottieInstance.isLoaded
+    ) {
+      dotLottieInstance.setBackgroundColor(config.backgroundColor);
+    }
+  });
+  // render config
+  createEffect(() => {
+    const dotLottieInstance = dotLottie();
+
+    if (!dotLottieInstance) return;
+
+    if (typeof config.renderConfig === 'object') {
+      dotLottieInstance.setRenderConfig(config.renderConfig);
+    }
+  });
+  // data
+  createEffect(() => {
+    const dotLottieInstance = dotLottie();
+
+    if (!dotLottieInstance) return;
+
+    if (typeof config.data == 'string' || config.data instanceof ArrayBuffer) {
+      dotLottieInstance.load({
+        data: config.data,
```

Comment:
On any config change, wouldn't that trigger the effect? I keep seeing in the example that play and load events are being triggered on changes that are not related to loading new data, such as changing the speed or frame interpolation.

I think we need to ensure that we're only triggering the effect for the specific property we want to react to, not for the entire configuration.

Reply:
Yes good catch, that was cause of de-structuring reactive props, `...config` in specific effect.
resolved on https://github.com/LottieFiles/dotlottie-web/pull/234/commits/179a084c8e4e108697fbe7aa626f06e24104b6a8